src/sidebar/app-controller.js

```diff
@@ -98,6 +100,11 @@ module.exports = function AppController(
 
   // Start the login flow. This will present the user with the login dialog.
   $scope.login = function () {
+    if (serviceConfig(settings)) {
```

Comment:
So this is a valid point. In principle, the login action is tied to a specific service. Other actions that we might want to notify the host page about (eg. "the sidebar opened") are not service-specific however.

We could move the `onLogin` callback to the entry in the `services` array. When we send an event back to the sidebar we would then need to include information about which service the user tried to log in to so that the appropriate service's login callback can be invoked.